sorc/chgres_cube.fd/surface.F90

```diff
@@ -164,26 +170,54 @@ subroutine surface_driver(localpet)
 !-----------------------------------------------------------------------
 
  if (convert_nst) call create_nst_esmf_fields
+ 
+!-----------------------------------------------------------------------
+! Adjust soil levels of input grid !! not implemented yet
+!-----------------------------------------------------------------------
+
+ call adjust_soil_levels(localpet)
 
 !-----------------------------------------------------------------------
 ! Horizontally interpolate fields.
 !-----------------------------------------------------------------------
 
  call interp(localpet)
-
+ 
+ if (.not. (vgtyp_from_climo .or. sotyp_from_climo)) then
+   !---------------------------------------------------------------------------------------------
+   ! Check for points where smois is too high to be a land point at a land point
+   !---------------------------------------------------------------------------------------------
+   !  
+   !call check_smois_water
+ endif
+ 
 !---------------------------------------------------------------------------------------------
-! Adjust soil/landice column temperatures for any change in elevation between the
+! Adjust soil/landice column temperatures for any change in elevation between
+! the
 ! input and target grids.
 !---------------------------------------------------------------------------------------------
 
  call adjust_soilt_for_terrain
-
+ 
 !---------------------------------------------------------------------------------------------
 ! Rescale soil moisture for changes in soil type between the input and target grids.
```

Comment:
FYI, The soil moisture rescaling assumes Noah LSM land states on input and output.  Does the HRRR data contain RUC LSM land states?  If so, this routine is not relevant.

Reply:
HRRR data is indeed RUC LSM data, but the soil types are the same classification system so I assumed this would still be relevant. However, I do believe the RUC LSM states (particularly how RUC treats soil type under land ice) are the cause of the soil moisture mismatches that you mentioned in a prior question.